scripts/utils/get-history-file.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# scripts/utils/get-history-file.sh
+#
+# Returns the session-aware history file path
+
+set -euo pipefail
+base_history_dir="${TMUX_COMPILE_HISTORY_DIR:-$HOME/.tmux-compile-history}"
+session_name="$(tmux display-message -p '#S' 2>/dev/null || echo default)"
```

Comment:
Using tmux '#S' could include characters that break pathnames (spaces, slashes, ..)

This need to be sanitized.